experimental/sql/dialects/ibis_dialect.py

```diff
@@ -275,6 +275,37 @@ def get(table: Region, metrics: Region, names: list[str]) -> 'Aggregation':
         })
 
 
+@irdl_op_definition
+class InnerJoin(Operation):
+  """
+  Models an ibis inner join of table `left` with table `right` and the condition
+  `predicates`. An empty region for `predicates` means that this operation is
+  just a cartesian product.
+
+  Example:
+
+  '''
+  ibis.inner_join() {
+    ibis.unbound_table() ...
+  } {
+    ibis.unbound_table() ...
+  } {
+    ibis.equals() ...
+  }
+  '''
+  """
+  name = "ibis.inner_join"
+
+  left = SingleBlockRegionDef()
+  right = SingleBlockRegionDef()
+  predicates = SingleBlockRegionDef()
+
+  @staticmethod
+  @builder
+  def get(left: Region, right: Region, predicates: Region) -> 'InnerJoin':
```

Comment:
If the names from left and right need to be distinct, wouldn't it be a good idea to assert that here?

Reply:
I am not sure we want to infer all the schemas here. This would require going over the entire tree. For the SSA-shaped dialect, this certainly makes sense tough!

Reply:
Ah, I tend to forget that such information isn't available in the tree dialect. OK, do it only in the SSA-shaped one, then.

Am I looking at this too quickly or is this really a downside of the tree-shaped dialect?

Reply:
It's certainly a downside of being more declarative (leave out details -> details might not check out later on). In our current pipeline, we can argue that ibis does these checks, so we are "fine", but obviously we don't want to rely on that. 

Notice also that if we had at the very beginning decided to model trees with yields, we could find the types there now potentially. Something like:
```
%0 : bag = rel_alg.project() {
  %1 : bag = rel_alg.table()...
  rel_alg.yield(%1)
}
```

That's certainly an interesting insight for the thesis :)

Reply:
Yep, indeed!